staging/src/k8s.io/client-go/transport/token_source.go

```diff
@@ -72,14 +93,20 @@ func NewCachedTokenSource(ts oauth2.TokenSource) oauth2.TokenSource {
 type tokenSourceTransport struct {
 	base http.RoundTripper
 	ort  http.RoundTripper
+	src  ResettableTokenSource
 }
 
 func (tst *tokenSourceTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	// This is to allow --token to override other bearer token providers.
 	if req.Header.Get("Authorization") != "" {
 		return tst.base.RoundTrip(req)
 	}
-	return tst.ort.RoundTrip(req)
+	resp, err := tst.ort.RoundTrip(req)
+	if apierrors.IsUnauthorized(err) && tst.src != nil {
```

Comment:
I expected this to check `err != nil && resp != nil && resp.StatusCode == 401`, not apierrors.IsUnauthorized

Reply:
`err == nil`? done.

Reply:
yeah, sorry